### PR TITLE
Update .NET 6 dependencies to preview 4

### PR DESCRIPTION
**Pull request by Eilon**
And in Blazor there are some code changes where I updated APIs and removed workarounds that are no longer needed.

### CODE REVIEW

File: eng/Microsoft.Extensions.targets
Comment (Eilon) [inline eng/Microsoft.Extensions.targets:3]: This version was chosen because it matches the chain of .NET 6 dependencies from the new version entry I added below. If these are mis-aligned, you'll get package downgrade errors.

Comment (Eilon) [reply eng/Microsoft.Extensions.targets:3]: To put it another way, there may very well be newer versions (of preview4) in the feeds, but this is the version that matches the newest version of the ASP.NET Core / Blazor packages from preview4.

**Review by Redth — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Comment by rmarinho** [comment]
Do we still need this @Eilon  ?

**Comment by Eilon** [comment]
We do, but it's blocked on the other BlazorWebView PR: https://github.com/dotnet/maui/pull/1021

It could be that by now enough other stuff in MAUI is updated to Preview4 that part of this PR isn't needed, but it's hard to say until the other PR is merged.

**Comment by Eilon** [comment]
@Redth / @jsuarezruiz - this PR is now rebased on the preview4 branch so once it passes CI, it should be good to merge.

**Pull request merged**
